graphql/schema.graphql

```diff
@@ -90,6 +90,80 @@ type Mutation {
   Edit schedule and Plug and Play Compnay 3's and 4's Algorithms
   """
   editCourse(input: EditCourseInput): Response!
+    @deprecated(reason: "Use updateSchedule instead")
+  """
+  Update an entire schdule.
+  """
+  updateSchedule(input: UpdateScheduleInput!): UpdateScheduleResponse!
+}
+
+input UpdateScheduleInput {
+  """
+  ID of the schedule to update. If not given, the current schedule will be updated.
+  """
+  id: ID
+  """
+  The updated courses
+  """
+  courses: [CourseSectionInput!]!
+  """
+  Whether to perform validation on the backend through algorithm 1.
+  """
+  skipValidation: Boolean!
+  """
+  Which algorithm to use. If COMPANY4 is selected then validation will not be performed regardless of skipValidation.
+  """
+  validation: Company!
+}
+
+input CourseSectionInput {
+  """
+  The course identifier
+  """
+  id: CourseUpdateInput!
+  """
+  How many hours per week a course takes
+  """
+  hoursPerWeek: Float!
+  """
+  Maximum capacity of the section
+  """
+  capacity: Int!
+  """
+  Section number for courses, eg: A01, A02
+  """
+  sectionNumber: String
+  """
+  Professor's info, if any professors are assigned. Usernames
+  """
+  professors: [String!]!
```

Comment:
Does this support writing any string as the new professor name or does the professor need to be a valid username? The way that I have setup right now is the user can edit to any string they want.

Reply:
So yeah... I don't think we can easily make it so prof info is updated so the idea here is you can pass the prof associated with a course section.
The array here is the `username` of the prof so when we receive back the data we look for the username name and attach it to the course section being edited.

Reply:
Ok that's fine, as long as we're still supporting editing Prof assignments in some fashion